### PR TITLE
Re-connect to dbus for each call

### PR DESCRIPTION
**Pull request by AdamWill**
By using the system bus and leaving our connection to it open
for the entire lifetime of the isotovideo process, we're setting
ourselves up to receive a lot of signals we don't deal with, and
just letting them accumulate. Since all we want to do is send a
signal and then go away, let's use a private socket, and
re-initialize the connection for each signal. The overhead of
doing this is not large, and we're sending small numbers of
signals per isotovideo process anyway (one signal per network on
startup, one signal at startup if a debug variable is set, and
one signal per network at end).

Thanks to @dvdhrm and @berrange for their help with this.

Related progress issue: https://progress.opensuse.org/issues/90872

### CODE REVIEW

**Review by Martchus — COMMENTED** [review]
If you've tested it I'd be generally fine with the approach. I suppose actually running `Net::DBus::Reactor` to process the signals is not easily possible because the backend is already occupied with its own event processing.

**Comment by Martchus** [comment]
And maybe an explaining comment wouldn't hurt. Otherwise this might be accidentally reverted next time someone thinks this could be optimized. Thinking about optimizing this, it would still be cool if the connection would be established before and closed after the loop instead within every iteration.

**Comment by AdamWill** [comment]
yeah, I had the same thought, but actually doing it seems like it would be a bit more awkward than the current simple code. And for the small number of calls we're doing I'm not totally sure it's worth it.

So far I've just tested that at least tests run and work properly with this change. I'll need to have our instances running it for a day or two to be sure whether it actually fixes the bug, because we encounter it kinda sporadically; I think our exact combination of number of workers on the host plus tests that tend to run simultaneously is very close to the limit, and it maybe gets pushed over if two or three updates get tests scheduled at the same time, or we test an update at the same time as a compose, or something like that.

**Comment by AdamWill** [comment]
eh, even implementing `_dbus_calls` is a bit awkward because of the error and return handling. I'm not sure it's really worth the effort. WDYT?

File: backend/qemu.pm [hunk 1]
```diff
@@ -135,10 +135,11 @@ sub stop_qemu {
 
 sub _dbus_do_call {
     my ($self, $fn, @args) = @_;
-    $self->{dbus}         ||= Net::DBus->system;
-    $self->{dbus_service} ||= $self->{dbus}->get_service("org.opensuse.os_autoinst.switch");
-    $self->{dbus_object}  ||= $self->{dbus_service}->get_object("/switch", "org.opensuse.os_autoinst.switch");
-    $self->{dbus_object}->$fn(@args);
+    my $bus         = Net::DBus->system(private => "true");
```
[the remark this answers is not in the document]
Comment (AdamWill) [reply backend/qemu.pm]: done.

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
not worth the effort :)

**Comment by Amrysliu** [comment]
I created a ticket because many jobs are incomplete, and the log message showed the related code in this pr. 
https://progress.opensuse.org/issues/91163

**Pull request merged**